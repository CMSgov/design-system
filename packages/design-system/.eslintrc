// These rules only apply to files relative to this directory, and allow us
// to apply a stricter set of rules to our package's JavaScript versus the
// documentation site's.
{
  "rules": {
    // We want a displayName for the generated React docs
    "react/display-name": "error",
    // Avoid exploits. If you need dangerouslySetInnerHTML, then temporarily
    // disable this rule in the script rather than removing it from here.
    "react/no-danger": "error"
  }
}
