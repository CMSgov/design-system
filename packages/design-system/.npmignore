**/__tests__/
**/__snapshots__/
**/helpers/
**/*.test.js
**/*.test.jsx
yarn.lock
.stylelintrc.js