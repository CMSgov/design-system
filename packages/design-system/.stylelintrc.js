module.exports = {
  extends: '@cmsgov/stylelint-config-design-system',
  rules: {
    /**
     * Class naming pattern for the core design system
     * ~~~~~~~~~~~~~~~~~~~~
     * Names:    ds-
     * Prefixes: l- c- u-
     * Pattern:  [NAMESPACE]-[PREFIX]-[BLOCK]__[ELEMENT]--[MODIFIER]
     * Examples: .ds-c-button--primary, .ds-c-card__title, .ds-u-text-underlined
     */
    'selector-class-pattern': /^(ds-)(l|c|u|)(-[a-z0-9]+)((--?|__)[a-z0-9]+)*$/,
  },
};
