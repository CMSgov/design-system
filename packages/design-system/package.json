{
  "name": "@cmsgov/design-system",
  "version": "3.1.0",
  "publishConfig": {
    "access": "public",
    "tag": "latest"
  },
  "description": "CMS Design System Sass/CSS and React assets",
  "repository": {
    "type": "git",
    "url": "https://github.com/CMSgov/design-system",
    "directory": "packages/design-system"
  },
  "license": "SEE LICENSE IN LICENSE.md",
  "main": "dist/components/index.js",
  "module": "dist/esnext/index.esm.js",
  "sideEffects": false,
  "types": "dist/types/index.d.ts",
  "files": [
    "dist",
    "src/components/**/*.{jsx,tsx}",
    "!src/**/*{.test,.spec,.d}.{jsx,tsx}"
  ],
  "dependencies": {
    "@popperjs/core": "^2.4.4",
    "@types/react": "^17.0.10",
    "@types/react-dom": "^17.0.10",
    "classnames": "^2.2.5",
    "core-js": "^3.6.5",
    "downshift": "^3.2.10",
    "ev-emitter": "^1.1.1",
    "focus-trap-react": "^6.0.0",
    "lodash": "^4.17.21",
    "prop-types": "^15.7.2",
    "react-aria-modal": "^2.11.1",
    "react-transition-group": "^4.4.2"
  },
  "peerDependencies": {
    "react": ">=16.0.0",
    "react-dom": ">=16.0.0"
  }
}
