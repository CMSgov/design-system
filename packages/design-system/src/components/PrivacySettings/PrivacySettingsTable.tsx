import React from 'react';
import { ChoiceList } from '../ChoiceList';
import { t } from '../i18n';

export interface PrivacySettingsProperty {
  settingsKey: string;
  translationKey: string;
  value: string;
}

export interface PrivacySettingsTableProps {
  domain: string;
  privacySettings: PrivacySettingsProperty[];
  setPrivacySetting: (key: string, value: string) => any;
}

export const PrivacySettingsTable = ({
  domain,
  privacySettings,
  setPrivacySetting,
}: PrivacySettingsTableProps) => {
  function renderToggle(settingsKey: string, value: string, category: string, description: string) {
    const choices = [
      {
        label: t('privacy.allow'),
        value: '0',
        checked: value === '0',
      },
      {
        label: t('privacy.dontAllow'),
        value: '1',
        checked: value === '1',
      },
    ];
    const label = <span className="ds-u-visibility--screen-reader">{category}</span>;
    return (
      <ChoiceList
        type="radio"
        choices={choices}
        onChange={(event) => {
          setPrivacySetting(settingsKey, event.target.value);
        }}
        className="ds-u-margin-top--0"
        label={label}
        aria-label={description}
        name={`cookie-${settingsKey}`}
      />
    );
  }

  function renderRow({ settingsKey, translationKey, value }: PrivacySettingsProperty) {
    const category = t(`privacy.${translationKey}.category`);
    const description = t(`privacy.${translationKey}.description`, { domain });
    return (
      <tr key={settingsKey}>
        <td data-title={t('privacy.category')}>{category}</td>
        <td data-title={t('privacy.description')}>{description}</td>
        <td data-title={t('privacy.status')}>
          {renderToggle(settingsKey, value, category, description)}
        </td>
      </tr>
    );
  }

  return (
    <table className="ds-c-table ds-c-table--borderless ds-c-privacy-settings-table">
      <thead>
        <tr>
          <th scope="col">{t('privacy.category')}</th>
          <th scope="col">{t('privacy.description')}</th>
          <th scope="col">{t('privacy.status')}</th>
        </tr>
      </thead>
      <tbody>{privacySettings.map(renderRow)}</tbody>
    </table>
  );
};

export default PrivacySettingsTable;
