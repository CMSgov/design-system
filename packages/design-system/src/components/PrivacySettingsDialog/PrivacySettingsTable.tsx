import React from 'react';
import { ChoiceList } from '../ChoiceList/index';
import { Table, TableHead, TableRow, TableCell, TableBody } from '../Table/index';
import { t } from '../i18n';

export interface PrivacySettingsProperty {
  settingsKey: string;
  translationKey: string;
  value: string;
}

export interface PrivacySettingsTableProps {
  domain: string;
  privacySettings: PrivacySettingsProperty[];
  setPrivacySetting: (key: string, value: string) => any;
}

export const PrivacySettingsTable = ({
  domain,
  privacySettings,
  setPrivacySetting,
}: PrivacySettingsTableProps) => {
  function renderToggle(settingsKey: string, value: string, category: string, description: string) {
    const choices = [
      {
        label: t('privacy.allow'),
        value: '0',
        checked: value === '0',
      },
      {
        label: t('privacy.dontAllow'),
        value: '1',
        checked: value === '1',
      },
    ];
    const label = <span className="ds-u-visibility--screen-reader">{category}</span>;
    return (
      <ChoiceList
        type="radio"
        choices={choices}
        onChange={(event) => {
          setPrivacySetting(settingsKey, event.target.value);
        }}
        className="ds-u-margin-top--0"
        label={label}
        aria-label={description}
        name={`cookie-${settingsKey}`}
      />
    );
  }

  function renderRow({ settingsKey, translationKey, value }: PrivacySettingsProperty) {
    const category = t(`privacy.${translationKey}.category`);
    const description = t(`privacy.${translationKey}.description`, { domain });
    return (
      <TableRow key={settingsKey}>
        <TableCell stackedTitle={t('privacy.category')}>{category}</TableCell>
        <TableCell stackedTitle={t('privacy.description')}>{description}</TableCell>
        <TableCell stackedTitle={t('privacy.status')}>
          {renderToggle(settingsKey, value, category, description)}
        </TableCell>
      </TableRow>
    );
  }

  return (
    <Table className="ds-c-privacy-settings-table" borderless stackable stackableBreakpoint="md">
      <TableHead>
        <TableRow>
          <TableCell>{t('privacy.category')}</TableCell>
          <TableCell>{t('privacy.description')}</TableCell>
          <TableCell>{t('privacy.status')}</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>{privacySettings.map(renderRow)}</TableBody>
    </Table>
  );
};

export default PrivacySettingsTable;
