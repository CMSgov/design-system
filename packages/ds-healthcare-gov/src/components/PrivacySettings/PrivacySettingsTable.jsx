import { ChoiceList } from '@cmsgov/design-system';
import PropTypes from 'prop-types';
import React from 'react';

export class PrivacySettingsTable extends React.PureComponent {
  renderToggle(settingsKey, value, category, description) {
    const { t, setPrivacySetting } = this.props;
    const choices = [
      {
        label: t('privacy.allow'),
        value: '0',
        checked: value === '0',
      },
      {
        label: t('privacy.dontAllow'),
        value: '1',
        checked: value === '1',
      },
    ];
    const label = <span className="ds-u-visibility--screen-reader">{category}</span>;
    return (
      <ChoiceList
        type="checkbox"
        choices={choices}
        onChange={(event) => {
          setPrivacySetting(settingsKey, event.target.value);
        }}
        className="ds-u-margin-top--0"
        label={label}
        aria-label={description}
        name={`cookie-${settingsKey}`}
      />
    );
  }

  renderRow({ settingsKey, translationKey, value }) {
    const { t } = this.props;
    const category = t(`privacy.${translationKey}.category`);
    const description = t(`privacy.${translationKey}.description`);
    return (
      <tr key={settingsKey}>
        <td data-title={t('privacy.category')}>{category}</td>
        <td data-title={t('privacy.description')}>{description}</td>
        <td data-title={t('privacy.status')}>
          {this.renderToggle(settingsKey, value, category, description)}
        </td>
      </tr>
    );
  }

  render() {
    const { t, privacySettings } = this.props;
    return (
      <table className="ds-c-table ds-c-table--borderless hc-c-privacy-settings-table">
        <thead>
          <tr>
            <th scope="col">{t('privacy.category')}</th>
            <th scope="col">{t('privacy.description')}</th>
            <th scope="col">{t('privacy.status')}</th>
          </tr>
        </thead>
        <tbody>{privacySettings.map((setting) => this.renderRow(setting))}</tbody>
      </table>
    );
  }
}

PrivacySettingsTable.propTypes = {
  t: PropTypes.func.isRequired,
  privacySettings: PropTypes.arrayOf(
    PropTypes.shape({
      settingsKey: PropTypes.string.isRequired,
      translationKey: PropTypes.string.isRequired,
      value: PropTypes.string.isRequired,
    })
  ).isRequired,
  setPrivacySetting: PropTypes.func.isRequired,
};

export default PrivacySettingsTable;
