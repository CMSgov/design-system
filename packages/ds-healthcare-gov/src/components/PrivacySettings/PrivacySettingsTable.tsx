import { ChoiceList } from '@cmsgov/design-system';
import React from 'react';

export interface PrivacySettingsProperty {
  settingsKey: string;
  translationKey: string;
  value: string;
}

export interface PrivacySettingsTableProps {
  t: (key: string) => string;
  privacySettings: PrivacySettingsProperty[];
  setPrivacySetting: (key: string, value: string) => any;
}

export const PrivacySettingsTable = (props: PrivacySettingsTableProps) => {
  const { t, privacySettings, setPrivacySetting } = props;

  function renderToggle(settingsKey: string, value: string, category: string, description: string) {
    const choices = [
      {
        label: t('privacy.allow'),
        value: '0',
        checked: value === '0',
      },
      {
        label: t('privacy.dontAllow'),
        value: '1',
        checked: value === '1',
      },
    ];
    const label = <span className="ds-u-visibility--screen-reader">{category}</span>;
    return (
      <ChoiceList
        type="radio"
        choices={choices}
        onChange={(event) => {
          setPrivacySetting(settingsKey, event.target.value);
        }}
        className="ds-u-margin-top--0"
        label={label}
        aria-label={description}
        name={`cookie-${settingsKey}`}
      />
    );
  }

  function renderRow({ settingsKey, translationKey, value }: PrivacySettingsProperty) {
    const category = t(`privacy.${translationKey}.category`);
    const description = t(`privacy.${translationKey}.description`);
    return (
      <tr key={settingsKey}>
        <td data-title={t('privacy.category')}>{category}</td>
        <td data-title={t('privacy.description')}>{description}</td>
        <td data-title={t('privacy.status')}>
          {renderToggle(settingsKey, value, category, description)}
        </td>
      </tr>
    );
  }

  return (
    <table className="ds-c-table ds-c-table--borderless hc-c-privacy-settings-table">
      <thead>
        <tr>
          <th scope="col">{t('privacy.category')}</th>
          <th scope="col">{t('privacy.description')}</th>
          <th scope="col">{t('privacy.status')}</th>
        </tr>
      </thead>
      <tbody>{privacySettings.map(renderRow)}</tbody>
    </table>
  );
};

export default PrivacySettingsTable;
