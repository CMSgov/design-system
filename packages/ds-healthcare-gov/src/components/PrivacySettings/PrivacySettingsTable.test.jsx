import { ChoiceList } from '@cmsgov/design-system';
import PrivacySettingsTable from './PrivacySettingsTable';
import React from 'react';
import { shallow } from 'enzyme';

const defaultProps = {
  t: (key) => key,
  privacySettings: [
    { settingsKey: 'key1', translationKey: 'setting1', value: '0' },
    { settingsKey: 'key2', translationKey: 'setting2', value: '1' },
    { settingsKey: 'key3', translationKey: 'setting3', value: '0' },
  ],
  setPrivacySetting: () => {},
};

function render(props) {
  props = Object.assign({}, defaultProps, props);
  const wrapper = shallow(<PrivacySettingsTable {...props} />);
  return { wrapper, props };
}

describe('<PrivacySettingsTable />', function () {
  it('renders the privacy settings table', () => {
    expect(render().wrapper).toMatchSnapshot();
  });

  it('calls the setPrivacySetting prop on Choice change', () => {
    const setPrivacySetting = jest.fn();
    const { wrapper } = render({ setPrivacySetting });
    const choiceList = wrapper.find(ChoiceList).first();
    choiceList.props().onChange({ target: { value: '1' } });
    expect(setPrivacySetting).toHaveBeenCalledWith('key1', '1');
  });
});
