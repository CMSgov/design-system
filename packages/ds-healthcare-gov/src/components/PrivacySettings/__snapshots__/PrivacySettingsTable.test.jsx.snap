// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PrivacySettingsTable /> renders the privacy settings table 1`] = `
<table
  className="ds-c-table ds-c-table--borderless hc-c-privacy-settings-table"
>
  <thead>
    <tr>
      <th
        scope="col"
      >
        privacy.category
      </th>
      <th
        scope="col"
      >
        privacy.description
      </th>
      <th
        scope="col"
      >
        privacy.status
      </th>
    </tr>
  </thead>
  <tbody>
    <tr
      key="key1"
    >
      <td
        data-title="privacy.category"
      >
        privacy.setting1.category
      </td>
      <td
        data-title="privacy.description"
      >
        privacy.setting1.description
      </td>
      <td
        data-title="privacy.status"
      >
        <ChoiceList
          aria-label="privacy.setting1.description"
          choices={
            Array [
              Object {
                "checked": true,
                "label": "privacy.allow",
                "value": "0",
              },
              Object {
                "checked": false,
                "label": "privacy.dontAllow",
                "value": "1",
              },
            ]
          }
          className="ds-u-margin-top--0"
          label={
            <span
              className="ds-u-visibility--screen-reader"
            >
              privacy.setting1.category
            </span>
          }
          name="cookie-key1"
          onChange={[Function]}
          type="checkbox"
        />
      </td>
    </tr>
    <tr
      key="key2"
    >
      <td
        data-title="privacy.category"
      >
        privacy.setting2.category
      </td>
      <td
        data-title="privacy.description"
      >
        privacy.setting2.description
      </td>
      <td
        data-title="privacy.status"
      >
        <ChoiceList
          aria-label="privacy.setting2.description"
          choices={
            Array [
              Object {
                "checked": false,
                "label": "privacy.allow",
                "value": "0",
              },
              Object {
                "checked": true,
                "label": "privacy.dontAllow",
                "value": "1",
              },
            ]
          }
          className="ds-u-margin-top--0"
          label={
            <span
              className="ds-u-visibility--screen-reader"
            >
              privacy.setting2.category
            </span>
          }
          name="cookie-key2"
          onChange={[Function]}
          type="checkbox"
        />
      </td>
    </tr>
    <tr
      key="key3"
    >
      <td
        data-title="privacy.category"
      >
        privacy.setting3.category
      </td>
      <td
        data-title="privacy.description"
      >
        privacy.setting3.description
      </td>
      <td
        data-title="privacy.status"
      >
        <ChoiceList
          aria-label="privacy.setting3.description"
          choices={
            Array [
              Object {
                "checked": true,
                "label": "privacy.allow",
                "value": "0",
              },
              Object {
                "checked": false,
                "label": "privacy.dontAllow",
                "value": "1",
              },
            ]
          }
          className="ds-u-margin-top--0"
          label={
            <span
              className="ds-u-visibility--screen-reader"
            >
              privacy.setting3.category
            </span>
          }
          name="cookie-key3"
          onChange={[Function]}
          type="checkbox"
        />
      </td>
    </tr>
  </tbody>
</table>
`;
